src/components/VolunteerAutomation/AttendClass.js

```diff
@@ -60,11 +60,21 @@ function AttendClass({
     dispatch(classActions.getClasses());
   }, [dispatch]);
 
-  const possibleClasses = data?.slice(0, numOfClassesToShow) || [];
+  const pathwayId = JSON.parse(
+    localStorage.getItem("volunteer_automation--state")
+  )?.pathwayId;
+
+  const classData = data?.filter((item) => item.pathway_v2 == pathwayId) || [];
+
+  const possibleClasses =
+    classData.length === 0
+      ? data?.slice(0, numOfClassesToShow) || []
+      : classData.length >= numOfClassesToShow
+      ? classData?.slice(0, numOfClassesToShow)
+      : classData;
```

Comment:
If `numOfClassesToShow` is greater than (or equal to) the `length` of `classData`, using `slice(0, numOfClassesToShow)` will simply cause the whole Array to be extracted (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#end), so it's not necessary to do the additional check `classData.length >= numOfClassesToShow`.  Additionally, it's not necessary to use the `?` since you already set `classData` to be the empty Array when `data` is `undefined` or `null`.
```suggestion
      : classData.slice(0, numOfClassesToShow);
```